shared/services/wallet/wallet.go

```diff
@@ -291,6 +292,27 @@ func (w *Wallet) Sign(serializedTx []byte) ([]byte, error) {
 	return signedData, nil
 }
 
+// Signs an arbitrary message using the wallet's private key
+func (w *Wallet) SignMessage(message string) ([]byte, error) {
+	// Get the wallet's private key
+	privateKey, _, err := w.getNodePrivateKey()
+	if err != nil {
+		return nil, err
+	}
+
+	fullMessage := fmt.Sprintf("\x19Ethereum Signed Message:\n%d%s", len(message), message)
+	hash := crypto.Keccak256Hash([]byte(fullMessage))
+
+	signedMessage, err := crypto.Sign(hash.Bytes(), privateKey)
+	if err != nil {
+		return nil, fmt.Errorf("Error signing message: %w", err)
+	}
+
+	// fix the ECDSA 'v' (see https://medium.com/mycrypto/the-magic-of-digital-signatures-on-ethereum-98fe184dc9c7#:~:text=The%20version%20number,2%E2%80%9D%20was%20introduced)
+	signedMessage[64] += 27
```

Comment:
based on my read of the above:
```markdown
In most implementations, [the ](https://github.com/ethereum/go-ethereum/issues/19751#issuecomment-504900739)[v](https://github.com/ethereum/go-ethereum/issues/19751#issuecomment-504900739)[ is just 0 or 1 internally](https://github.com/ethereum/go-ethereum/issues/19751#issuecomment-504900739), but 27 was added as arbitrary number for signing Bitcoin messages and Ethereum adapted that as well.
```
Shouldn't this be:
`signedMessage[64] = 27`? To guard against changes in the underlying implementation (keyword above being "most" implementations.

Reply:
I would stick to the spec and keep adding 27 to this bit. I think using a fixed value makes it incompatible with ethers.js signatures for instance.